Scripts/Singletons/SceneSwapper.gd

```diff
@@ -76,13 +78,24 @@ func load_back_button(level_select):
 
 
 func load_lvl_buttons(level_select):
+	var completedTheme = load("res://Scenes/Stage_Select/LevelCompletedButton.tres")
+	var stage_type = level_select.name.replace("Select", "")
 	var btn_nodes = get_tree().get_nodes_in_group("level_buttons")
 	var btn_count = 0
 	
 	var x_pos = X_START
 	var y_pos = Y_START
 	
 	for btn in btn_nodes:
+		#Set button color
+		var level_number = btn.text
```

Comment:
scene_array was renamed to scene_order and only contains the keys instead of the level paths (which is honestly probably better for this implementation too?). I don't think it will affect any of your code too much we'll just have to update these lines a bit

Reply:
I meant to comment on the lines below this one but ykwim

Reply:
can I pull from LevelDesign to make sure the rework of this works, or is that branch still a lil funky?